### PR TITLE
Tests for encoding of macro parameters

### PR DESCRIPTION
**Pull request by mrosanes**
Insert tests to check the correct encoding/decoding of macro
parameters.

Using insertTest decorator, different parameter combinations
(including repeat parameters) are tested.

### CODE REVIEW

File: src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+##############################################################################
+
+"""Resource file containing parameter definitions and related
+input and expected parameters, to be used in test_macro for testing
+encoding and decoding of macro parameters."""
+
+
+################################################################################
+
+params_definition_move = [
```
Comment (reszelaz) [inline src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py]: I would reference in the variable names with the macro examples located in `sardana.macroserver.macros.examples.parameters`. So `pt0_param_def`, `pt01_param_def`, ...

Comment (reszelaz) [reply src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py]: Apart of that I would make a dedicated module called: `paramdef` where I would place all these definitions together with the one already defined in `test_macro` module.

File: src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+         'default_value': "mot99",
+         'type': 'Moveable'},
+        {'name': 'pos',
+         'default_value': 100,
+         'type': 'Float'}
+        ]}
+    ]
+
+
+move_parameters_1 = [[["mot01", "0"]]]
```
Comment (reszelaz) [inline src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py]: Personally I would keep the parameter values close to the `insertTest` calls so in the test module. I think that we will gain the readability. But treat is just as a suggestion.

File: src/sardana/taurus/core/tango/sardana/test/test_macro_paramrepeat.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
```
Comment (reszelaz) [inline src/sardana/taurus/core/tango/sardana/test/test_macro_paramrepeat.py]: I would merge this module with the already existing one `test_macro`.

**Review by reszelaz — CHANGES_REQUESTED** [review]

File: src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py (hunk 1 above)
Comment (mrosanes) [reply src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py]: Done

File: src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py (hunk 2 above)
Comment (mrosanes) [reply src/sardana/taurus/core/tango/sardana/test/res/params_definitions.py]: Done

File: src/sardana/taurus/core/tango/sardana/test/test_macro_paramrepeat.py (hunk 3 above)
Comment (mrosanes) [reply src/sardana/taurus/core/tango/sardana/test/test_macro_paramrepeat.py]: Done

**Review by reszelaz — APPROVED** [review]

**Pull request merged**